src/command_modules/azure-cli-sf/tests/test_sf_commands.py

```diff
@@ -0,0 +1,154 @@
+from azure.cli.testsdk.base import execute
+from azure.cli.testsdk.preparers import AbstractPreparer
+
+
+class SelectNoSecClusterPreparer(AbstractPreparer):
+    def __init__(self, parameter_name="endpoint",
+                 endpoint="http://127.0.0.1:10550",
+                 env_variable_name="AZURE_CLI_SF_ENDPOINT"):
+        # Name randomization unnecessary
+        super(SelectNoSecClusterPreparer, self).__init__("test", 10)
+        self.endpoint = endpoint
+        self.parameter_name = parameter_name
+        self.env_variable_name = env_variable_name
+
+    def create_resource(self, _, **kwargs):
+        # Omit name here since there is no randomization required
+        endpoint = os.environ.get(self.env_variable_name, self.endpoint)
+        print("endpoint {}".format(endpoint))
+        template = "az sf cluster select --endpoint {}"
+        execute(template.format(endpoint))
+        return {self.parameter_name: endpoint}
+
+    def remove_resource(self, _):
+        # Do nothing to unset connection
+        return
+
+class ServiceFabricTests(ScenarioTest):
+
+    # Application tests
+
+    @SelectNoSecClusterPreparer()
+    def test_good_system_app_health(self):
+        self.cmd("az sf application health --application-id \"System\"", checks=[
+            JMESPathCheck("name", "fabric:/System"),
+            JMESPathCheckExists("aggregatedHealthState")
+        ])
+
+    @SelectNoSecClusterPreparer()
+    def test_good_none_app_type(self):
+        self.cmd("az sf application type", checks=[NoneCheck()])
+
+    @SelectNoSecClusterPreparer()
+    def test_good_none_app_list(self):
+        self.cmd("az sf application list", checks=[
+            JMESPathCheck("items", [])
+        ])
+
+    # Service tests
+
+    @SelectNoSecClusterPreparer()
+    def test_good_system_service_list(self):
+        self.cmd("az sf service list --application-id \"System\"", checks=[
+            JMESPathCheck(
+                "items[? id == `\"System/ClusterManagerService\"`].name | [0]",
+                "fabric:/System/ClusterManagerService"),
+        ])
+
+    @SelectNoSecClusterPreparer()
+    def test_good_system_service_app_name(self):
+        self.cmd("az sf service application-name --service-id \
+        System/ClusterManagerService", checks=[
+            JMESPathCheck("id", "System"),
+            JMESPathCheck("name", "fabric:/System")
+        ])
+
+    @SelectNoSecClusterPreparer()
+    def test_good_system_service_health(self):
+        self.cmd("az sf service health --service-id \
+        System/ClusterManagerService", checks=[
+            JMESPathCheck("name", "fabric:/System/ClusterManagerService"),
+            JMESPathCheckExists("partitionHealthStates"),
+            JMESPathCheckExists("healthEvents")
+        ])
+
+    @SelectNoSecClusterPreparer()
+    def test_good_resolve_system_service(self):
+        self.cmd("az sf service resolve --service-id \
+            System/FailoverManagerService")
```

Comment:
Not for now, since we cannot predict what the partition will resolve to, our best bet is just to see that we get some not empty response.